bikram.md

```diff
@@ -0,0 +1,95 @@
+
+3	Masking HTTP Headers:
+Server versioning information or any other sensitive information from the HTTP headers should be removed/masked according to industry best practices. This prevents any form of targeted attacks since the vulnerabilities are mostly specific to the vendors.
+
+4	Session management:
+RESTful web services should use session-based authentication, either by establishing a session token via a POST or by using an API key (client_id + client_secret) as a POST body argument or as a cookie. Usernames, passwords, session tokens, API keys, and sensitive information should not appear in the URL, as this can be captured in web server logs, which makes them intrinsically valuable.
+
+5	Protect HTTP methods:
+RESTful API often use GET (read), POST (create), PUT (replace/update) and DELETE (to delete a record). Not all of these are valid choices for every single resource collection, user, or action. Make sure the incoming HTTP method is valid for the session token/API key and associated resource collection, action, and record.
+
+6	HTTP Return Code:
+HTTP defines status code. While designing a REST API, don't just use 200 for success or 404 for error. Every error message needs to be customized as to not reveal any unnecessary information. Here are some guidelines to consider for each REST API status return code. Proper error handle may help to validate the incoming requests and better identify the potential security risks. 
+•	200 OK - Response to a successful REST API action. The HTTP method can be GET, POST, PUT, PATCH or DELETE.
+•	400 Bad Request - The request is malformed, such as message body format error.
+•	401 Unauthorized - Wrong or no authentication ID/password provided.
+•	403 Forbidden - It's used when the authentication succeeded but authenticated user doesn't have permission to the request resource
+•	404 Not Found - When a non-existent resource is requested
+•	405 Method Not Allowed - The error checking for unexpected HTTP method. For example, the RestAPI is expecting HTTP GET, but HTTP PUT is used.
+•	429 Too Many Requests - The error is used when there may be DOS attack detected or the request is rejected due to rate limiting
+
+7	Input Validation:
+Everything you know about input validation applies to RESTful web services, but add 10% because automated tools can easily fuzz your interfaces for hours on end at high velocity. Help the user input high quality data into your web services, such as ensuring a Zip code makes sense for the supplied address, or the date makes sense. If not, reject that input. Also make sure that the output encoding is very strong for your application. Some other specific forms of input validations need to be implemented:
```

Comment:
> Everything you know about input validation applies

I would have little expectation from the reader here. Maybe elaborate a bit more (or link to an external reading)?